### DISCUSSION TITLE
Multiphase Invasion Simulation

### DISCUSSION THREAD

**Discussion by falvlima** [PMEAL/OpenPNM#2322] [Q&A]
Hi. I am trying to perform a simulation where I have the porous media previously saturated with fluid n.1  (o partially saturated, and also air). 
Then I would like to inject a fluid2 and check the invasion sequence as well as other properties available in the MixedInvasionPercolation algorithm.
I've thought about performing the first invasion and then saving pores and throats hydraulic properties for the first fluid (entry pressure, hc, etc) and then performing a second invasion using the second fluid.
Would it be possible with this algorithm? If yes, how could I do it?
I have checked the  "Part C: Mixed (Invasion) Percolation" example, but in this case, pores filled in the first invasion would remain apparently inaccessible. I just would like to set the initial state from the first invasion but, for example, if a high pressure (pc) is applied, I could check if the defending fluid would be expulsed or not.
Thanks in advance.
Perhaps it is a trivial question, but I am a beginner user.

**Comment by jgostick** [comment]
If I understand correctly, what you are proposing (expulsion of residual phase) is quite difficult and requires physics that we have not implemented in openpnm.  Residual phase can be present during a secondary injection and it will help the invasion phase by providing 'free connections', but causing it to move is not possible.  For instance applyin a high capillary pressure will just apply the same pressure to all sides of the droplet, so it won't move.  To make it move you need a pressure *gradient* across the droplet.  In principle you could run a StokesFlow simulation on the domain and check if the pressure different acting on the droplet is strong enough to cause motion, but then you'd need to transfer this information back to the invasion percolation algorithm.  Basically, it's complicated and is beyond what openpnm can do, and I don't see us adding anything like that in the near future.

**Reply by falvlima** [reply]
Hi, Jeff! Thanks a lot for the quick reply! We have some experimental data from some fast tomo and I would like to compare the invasion to the "populated" porous media. I will try to "play" with both algorithms the following days and check if we can get the expected results from that. Anyway, apparently, the code will be pretty helpful, even if we need to simplify the simulation.
